jax/interpreters/masking.py

```diff
@@ -89,31 +86,34 @@ def mask_subtrace(master, in_vals, shape_exprs):
   out_vals, out_shapes = unzip2((t.val, t.shape_expr) for t in out_tracers)
   yield out_vals, out_shapes
 
-def ensure_poly(p):
-  if isinstance(p, Poly):
+def to_index(x):
+  """Like operator.index, but allowing polymorphic dimensions.
+  Not implemented as `Poly.__index__`, since operator.index only allows ints."""
+  return x if type(x) is Poly else op.index(x)
```

Comment:
We consolidated similar logic into `canonicalize_shape`/`_canonicalize_dimension` in `core.py`. Could you make use of those here?

(It also helps avoid some potential circular dependency issues.)

Reply:
Ok, nice! I just removed `to_index`, `canonicalize_shape` is now used everywhere.